test/common/test_common.cpp

```diff
@@ -284,6 +284,128 @@ TEST (PCL, PointCloud)
   EXPECT_FALSE (cloud.isOrganized ());
   EXPECT_EQ (cloud.width, 2);
   EXPECT_EQ (&new_point, &cloud.back ());
+
+  cloud.resize (640*360);
+  EXPECT_EQ (cloud.width, 640*360);
+  EXPECT_EQ (cloud.height, 1);
+
+  cloud.resize (640, 480);
+  EXPECT_EQ (cloud.width, 640);
+  EXPECT_EQ (cloud.height, 480);
+
+  cloud.resize (640*360, PointXYZ (1, 1, 1));
+  EXPECT_EQ (cloud.width, 640*360);
+  EXPECT_EQ (cloud.height, 1);
+
+  cloud.resize (640, 480, PointXYZ (1, 1, 1));
+  EXPECT_EQ (cloud.width, 640);
+  EXPECT_EQ (cloud.height, 480);
+
+  cloud.assign (640*360, PointXYZ (1, 1, 1));
+  EXPECT_EQ (cloud.width, 640*360);
+  EXPECT_EQ (cloud.height, 1);
+
+  cloud.assign(640, 480, PointXYZ (1, 1, 1));
+  EXPECT_EQ (cloud.width, 640);
+  EXPECT_EQ (cloud.height, 480);
+
+  std::vector<PointXYZ> pointVec;
+  pointVec.resize (640*360, PointXYZ (2, 3, 4));
+  cloud.assign (pointVec.begin(), pointVec.end());
+  EXPECT_EQ (cloud.width, 640*360);
+  EXPECT_EQ (cloud.height, 1);
+
+  pointVec.resize(640*480, PointXYZ (3, 4, 5));
+  cloud.assign (pointVec.begin(), pointVec.end(), 640);
+  EXPECT_EQ (cloud.width, 640);
+  EXPECT_EQ (cloud.height, 480);
+
+  pointVec.resize (640*480, PointXYZ (7, 7, 7));
+  cloud.assign (pointVec.begin(), pointVec.end(), 460);
+  EXPECT_EQ (cloud.width, 640*480);
+  EXPECT_EQ (cloud.height, 1);
+
+  //initializer_list
+  cloud.assign ({PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5),
+                PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5)});
+  EXPECT_EQ (cloud.width, 10);
+  EXPECT_EQ (cloud.height, 1);
+
+  cloud.assign ({PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5),
+                PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5),
+                PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5)}, 5);
+  EXPECT_EQ (cloud.width, 5);
+  EXPECT_EQ (cloud.height, 3);
+
+  cloud.assign ({PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5),
+                PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5),
+                PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5)}, 6);
+  EXPECT_EQ (cloud.width, 15);
+  EXPECT_EQ (cloud.height, 1);
+
+  cloud.push_back (PointXYZ (3, 4, 5));
+  EXPECT_EQ (cloud.width, 16);
+  EXPECT_EQ (cloud.height, 1);
+
+  pointVec.resize (640*480, PointXYZ (1, 1, 1));
+  cloud.assign (pointVec.begin(), pointVec.end(), 640);
+  cloud.transient_push_back (PointXYZ(2, 2, 2));
+  EXPECT_EQ (cloud.width, 640);
+  EXPECT_TRUE (cloud.isOrganized ());
+
+  auto& new_pointXYZ = cloud.transient_emplace_back (3, 3, 3);
+  EXPECT_TRUE (cloud.isOrganized ());
```

Comment:
Any reason for not testing the width here?

Reply:
yeah, we need to, I probably overlooked it.